modules.js

```diff
@@ -198,9 +258,57 @@ const createModule = async (
       is_available_for_new_content: false,
     };
 
-    fs.writeJSONSync(dest, metaData, { spaces: 2 });
+    if (!reactType) {
+      fs.writeJSONSync(dest, metaData, { spaces: 2 });
+    } else {
+      const globalImportString = getInternalVersion
+        ? 'import "./global-samplejsr.css";'
+        : '';
+      const defaultconfigString = getInternalVersion
+        ? `export const defaultModuleConfig = {
+  moduleName: "sample_jsr",
+  version: 0,
+};
+    `
+        : '';
+
+      fs.readFile(`${destPath}/index.tsx`, 'utf8', function(err, data) {
+        if (err) {
+          logger.error(
+            i18n(`${i18nKey}.errors.fileReadFailure`, {
+              path: `${dest}/index.tsx`,
+            })
+          );
+          return;
+        }
+
+        const result = data
+          .replace(/\/\* import global styles \*\//g, globalImportString)
+          .replace(/\/\* Default config \*\//g, defaultconfigString);
+
+        fs.writeFile(`${destPath}/index.tsx`, result, 'utf8', function(err) {
+          if (err) return console.log(err);
+        });
+
+        fs.appendFile(
+          `${dest}/index.tsx`,
+          'export const meta = ' + JSON.stringify(metaData, null, ' '),
+          err => {
+            if (err) {
+              logger.error(
+                i18n(`${i18nKey}.errors.failedToWrite`, {
+                  path: `${dest}/index.tsx`,
+                })
+              );
+              return;
+            }
+          }
+        );
+      });
+    }
```

Comment:
This could be moved to another function to keep the write module Meta function cleaner, but I'm leaving here for now. Wondering if there is a cleaner way to do this (read the index file, make some string replacements based on regex, and write back to it.

Reply:
I think it would be best if this logic was all pulled out into a separate function, yeah. Something like Read file -> pass contents to a transform function -> write file back in this function. Will make the transform much easier to test if it just takes in file contents rather than being intermingled w/ the rest of the createModule code.